File: easy/array/first_unique_char/first_unique_char.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        char_to_index = [0]*26
        for c in s:
            char_to_index[ord(c)-ord('a')] += 1

        for i, c in enumerate(s):
            if char_to_index[ord(c)-ord('a')] == 1:
                return i

        return -1


s = Solution()
print("Solution 1 : ", s.firstUniqChar("leetcode"))
print("Solution 2 : ", s.firstUniqChar("loveleetcode"))

File: easy/array/first_unique_char/first_unique_char.go
package main

import "fmt"

func firstUniqChar(s string) int {
	charToIndex := make([]int, 26)
	for _, c := range s {
		charToIndex[c-'a']++
	}

	for i, c := range s {
		if charToIndex[c-'a'] == 1 {
			return i
		}
	}
	return -1
}

func main() {
	fmt.Println("Solution 1 : ", firstUniqChar("leetcode"))
	fmt.Println("Solution 2 : ", firstUniqChar("loveleetcode"))
}
